src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/SentencesExtractionService.kt

```diff
@@ -62,6 +63,15 @@ class SentencesExtractionService {
     return sentenceStream.allMatch { sentence -> sentence.concurrentSentences.isEmpty() }
   }
 
+  fun allSentencesContainType(sentences: MutableList<Sentence>, sentenceType: SentenceType): Boolean {
+    for (sentence in sentences) {
```

Comment:
We could modify this `for loop` to be more functional; e.g. something like this
`return sentences.any { sentence -> sentences.any { innerSentence -> doSentencesOverlap(sentence, innerSentence)} }`